_posts/2021-04-22-release-1.13.0.md

```diff
@@ -0,0 +1,374 @@
+
+It is desirable to provide better visibility into the distribution of CPU resources while executing
+user code. One of the most visually effective means to do that are Flame Graphs. They allow to
+easily answer question like:
+Which methods are currently consuming CPU resources? How does consumption by one method compare to
+the others? Which series of calls on the stack led to executing a particular method? Flame Graphs
+are constructed by sampling stack traces a number of times. Every method call is represented by a
+bar, where the length of the bar is proportional to the number of times it is present in the
+samples. In order to prevent unintended impacts on production environments, Flame Graphs are
+currently available as an opt-in feature that needs to be enabled in the configuration. Once enabled
+they are accessible via a new component in the UI at the level of the selected operator:
+
+<figure style="align-content: center">
+  <img src="{{ site.baseurl }}/img/blog/2021-04-xx-release-1.13.0/7.png" style="display: block; margin-left: auto; margin-right: auto; width: 600px"/>
+</figure>
+
+### Access Latency Metrics for State
+
+State interactions are a crucial part of the majority of data
+pipelines. Especially in case of using RocksDB they might be rather IO intensive and therefore they
+play an important role in the overall performance of the pipeline. Therefore, it is important to be
+able to get insights into what is going on under the hood. To provide more insights, we exposed
+latency tracking metrics.
+
+The metrics are disabled by default, but you can enable them using the
+`state.backend.rocksdb.latency-track-enabled` option.
+
+### Unified binary savepoint format
+
+All available state backends are forced to produce a single common unified binary format for their
+savepoints. This means that savepoints are now mutually interchangeable. You are no longer locked
+into the first state backend you chose when starting your application for the first time. It makes
+it easier to start with Heap Backend and switch later on to RocksDB, if JVM Heap becomes too full (
+which you usually see when the GC times start to go up too much).
+
+### Support user-specified pod templates for Active Kubernetes Deployments
+
+The native Kubernetes deployment received an important update that it supports custom pod templates.
+Flink from now on allows users to define the JobManager and TaskManager pods via template files.
+This allows to support advanced features that are not supported by Flink Kubernetes config options
+directly. Major Observability Improvements
+
+What runs on Flink are often critical workloads with SLAs, so it is important to have the right
+tools to understand what is happening inside the applications.
+
+If your application does not progress as expected, the latency is higher or the throughput lower
+than you would expect, these features help you figure out what is going on.
+
+### Unaligned Checkpoints - Production Ready
```

Comment:
@pnowojski Could you help me fill this in?

Reply:
```
After fixing two more bugs related to Unaligned Checkpoints that were discovered in Flink 1.12.2,
upgrade to Flink 1.12.3 or Flink 1.13.0 is recommended for all Unaligned Checkpoints. Combined
with support of rescaling from Unaligned Checkpoints, this feature is now considered production ready.
```
?

Reply:
I'd not mention 1.12.3 here and focus more on features.

> 
> Unaligned checkpoints now support adaptive triggering with timeouts which will only perform unaligned checkpoint in case of backpressure and use aligned checkpoint otherwise. Thus, checkpoint times become reliable in all situation while keeping extra state to a minimum.
> 
> Further, you can now rescale from unaligned checkpoints to provide more resources in case of sudden spikes (together with reactive mode). This feature makes it easier to catch up and reduce backpressure in a timely fashion.
> 
> Flink 1.13 brings together all features that the community initially envisioned for unaligned checkpoints. Together with all bugfixes that happened in 1.12, we generally encourage the use of unaligned checkpoints for all applications with potential backpressure.

Reply:
I'll go with Arvid's proposal. I hope that's fine @pnowojski ;)